From 866d2738baf409f245eecef8cabed50aaf401770 Mon Sep 17 00:00:00 2001
From: Murat Sezgin <quic_msezgin@quicinc.com>
Date: Mon, 18 Apr 2022 17:37:04 -0700
Subject: [PATCH] net: skbuff: Fix compile warning when skb recycler is
 disabled

Move the variable inside the CONFIG_SKB_RECYCLER
option check.

Change-Id: I53e38159e3ca131228e8ad5b5d0be263848c2a70
Signed-off-by: Murat Sezgin <quic_msezgin@quicinc.com>
---
 net/core/skbuff.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -486,7 +486,6 @@ struct sk_buff *__netdev_alloc_skb(struc
 				   unsigned int length, gfp_t gfp_mask)
 {
 	struct sk_buff *skb;
-	struct skb_shared_info *shinfo;
 	unsigned int len = length;
 
 #ifdef CONFIG_SKB_RECYCLER
@@ -510,6 +509,7 @@ struct sk_buff *__netdev_alloc_skb(struc
 	 * Change the skb end to keep it in sync with the truesize.
 	 */
 	if (length + NET_SKB_PAD < SKB_WITH_OVERHEAD(PAGE_SIZE)) {
+		struct skb_shared_info *shinfo;
 		skb->truesize = SKB_TRUESIZE(SKB_DATA_ALIGN(length + NET_SKB_PAD));
 		skb->end = skb->tail + SKB_DATA_ALIGN(length + NET_SKB_PAD);
 
